Homepage: http://hg.punctweb.ro

--- a/dwm.c	2012-06-24 11:03:55.996900806 +0300
+++ b/dwm.c	2012-06-24 11:03:55.996900806 +0300
@@ -863,6 +863,8 @@ drawtext(const char *text, unsigned long
 
 void
 enternotify(XEvent *e) {
+	if(clicktofocus) return;
+
 	Client *c;
 	Monitor *m;
 	XCrossingEvent *ev = &e->xcrossing;
@@ -1253,6 +1255,8 @@ monocle(Monitor *m) {
 
 void
 motionnotify(XEvent *e) {
+	if(clicktofocus) return;
+
 	static Monitor *mon = NULL;
 	Monitor *m;
 	XMotionEvent *ev = &e->xmotion;
