Homepage: http://hg.punctweb.ro

--- a/dwm.c	2012-11-06 23:13:29.727844755 +0200
+++ b/dwm.c	2012-11-06 23:13:29.727844755 +0200
@@ -866,6 +866,8 @@ drawtext(const char *text, XftColor col[
 
 void
 enternotify(XEvent *e) {
+	if(clicktofocus) return;
+
 	Client *c;
 	Monitor *m;
 	XCrossingEvent *ev = &e->xcrossing;
@@ -1234,6 +1236,8 @@ monocle(Monitor *m) {
 
 void
 motionnotify(XEvent *e) {
+	if(clicktofocus) return;
+
 	static Monitor *mon = NULL;
 	Monitor *m;
 	XMotionEvent *ev = &e->xmotion;
