Homepage: http://hg.punctweb.ro

--- a/dwm.c	2012-09-19 10:20:15.015714491 +0300
+++ b/dwm.c	2012-09-19 10:20:15.018714491 +0300
@@ -868,6 +868,8 @@ drawtext(const char *text, unsigned long
 
 void
 enternotify(XEvent *e) {
+	if(clicktofocus) return;
+
 	Client *c;
 	Monitor *m;
 	XCrossingEvent *ev = &e->xcrossing;
@@ -1258,6 +1260,8 @@ monocle(Monitor *m) {
 
 void
 motionnotify(XEvent *e) {
+	if(clicktofocus) return;
+
 	static Monitor *mon = NULL;
 	Monitor *m;
 	XMotionEvent *ev = &e->xmotion;
