Homepage: http://hg.punctweb.ro

--- a/dwm.c	2012-06-24 11:03:56.066900806 +0300
+++ b/dwm.c	2012-06-24 11:03:56.076900806 +0300
@@ -1653,7 +1653,7 @@ setmfact(const Arg *arg) {
 	if(!arg || !selmon->lt[selmon->sellt]->arrange)
 		return;
 	f = arg->f < 1.0 ? arg->f + selmon->mfacts[selmon->curtag] : arg->f - 1.0;
-	if(f < 0.1 || f > 0.9)
+	if(f < 0.05 || f > 0.95)
 		return;
 	selmon->mfacts[selmon->curtag] = f;
 	arrange(selmon);
